test/drenv/providers/lima/k8s.yaml

```diff
@@ -120,6 +120,28 @@ provision:
       EOF
       systemctl restart containerd
 
+  # Pre-pull kubeadm images to improve probing performance
+  # See <https://github.com/lima-vm/lima/pull/3082>
+  - mode: system
+    script: |
+      #!/bin/bash
+      set -eux -o pipefail
+      test -e /etc/kubernetes/admin.conf && exit 0
+      command -v kubeadm >/dev/null 2>&1 || exit 0
+      # Ramen: Use local registry for k8s images
+      if [ -n "{{.Param.LOCAL_REGISTRY}}" ]; then
+        IMAGE_REPOSITORY="{{.Param.LOCAL_REGISTRY}}/k8s"
+      else
+        IMAGE_REPOSITORY=""
+      fi
+      # Pre-pull images using kubeadm config
+      cat <<EOF >kubeadm-images-config.yaml
+      kind: ClusterConfiguration
+      apiVersion: kubeadm.k8s.io/v1beta3
+      imageRepository: "$IMAGE_REPOSITORY"
+      EOF
+      kubeadm config images pull --config kubeadm-images-config.yaml
```

Comment:
Using --config for pulling the image from the local registry is nice, but how does it improve probing?

The upstream change added a separate probing script to show progress when the images are pulled, since pulling the images can take lot of time with slow network. Probing does not help us since we don't show the probing logs in drenv. So this change does help with drenv.

When using local registry the images are pulled in 10 seconds with the current code. This change makes sense only if it improves the time in a significant way.

When using remote registry the time can be much longer. Does this change improve the pull time?